feat: Implement Event Management CRUD

rasfaxo opened the pull request:
## Description

Implements Event Management CRUD with admin-only creation, public browsing at `/events`, and detail pages at `/events/$eventId`. Uses TanStack Middleware `requireAdmin` for reusable authorization checks instead of per-function session validation. Banner images use URL text input (Unsplash links) as MVP approach before photo upload feature.

**Key improvements:**


- Admins can create events via /admin/events/new (form uses shared UI components and live preview)
- Public and admin users can browse events at /events (admin sees all, public sees only public events)
- Event detail page with hero banner and consistent layout
- Unlisted events are visible only to admins, with a badge indicator
- Seed script includes placeholder images, an unlisted event, and an event without an image


**Reviewers should pay attention to:**
- Middleware pattern implementation in `src/lib/middleware.ts`
- Schema validation using Zod in `src/contracts/events.contract.ts`
- Server function authorization flow with `.middleware([requireAdmin])`


## Issue

Closes #21 

## Testing

- [x] `bun run check`
- [x] `bun run db:reset` _(when schema or seeds change)_ - **N/A**: No schema changes, uses existing `event` table
- [ ] `bun run build` _(optional, recommended when build/config tooling changes)_

## Checklist

- [x] Code is up-to-date with the `main` branch
- [x] Commit messages follow Conventional Commits
- [x] Screenshots or notes included for visual changes
- [ ] Documentation and `.env.example` updated when required
- [x] Schema changes include migration via `bun run db:generate` _(see [DATABASE.md](docs/DATABASE.md))_ - **N/A**: Uses existing schema


## Screenshots

1. `/events` grid layout
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/a83d773b-66bf-45aa-aff7-64fd25086979" />



2. `/events/$eventId` detail page 
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/aca9cb72-5cfc-422e-aaf3-7810af8f1aa5" />


4. `/admin/events/new` form UI
<img width="1920" height="1080" alt="image" src="https://github.com/user-attachments/assets/ab540d5d-8adc-4e33-8cac-7491087f2380" />


6. Mobile responsive views
<img width="430" height="871" alt="image" src="https://github.com/user-attachments/assets/59349efd-7a27-414d-b159-9dfc09b9ccaf" />


7.Mobile Create Event 
<img width="430" height="871" alt="image" src="https://github.com/user-attachments/assets/60f9c243-a1fc-4c76-8b8e-7660706ae3a7" />

rasfaxo committed:
feat(auth): create admin middleware for role-based access control

- Add requireAdmin middleware for admin-only access
- Implement using TanStack createMiddleware pattern
- Reusable across admin event CRUD operations

rasfaxo committed:
feat(events): add event validation contracts with Zod schemas

- Add atomic schemas: eventIdSchema, eventNameSchema, eventDescriptionSchema
- Add location, image URL, and visibility schemas
- Add composite schemas: eventCreateSchema, eventUpdateSchema, eventListQuerySchema
- Add eventRecordSchema for database records
- Export TypeScript types: EventCreate, EventUpdate, EventListQuery, EventRecord
- Provide type-safe validation for all event operations

rasfaxo committed:
feat(events): create public event listing page at /events

- Build grid layout with event cards (3 columns on desktop, responsive)
- Display event banner, name, location, and date on each card
- Filter to show only public visibility events
- Use TanStack Router loader for server-side data fetching

rasfaxo committed:
feat(events): create event detail page at /events/\$eventId

- Build hero banner section with gradient overlay
- Display full event metadata: date, time, location, visibility
- Show event description with proper formatting
- Add back navigation button
- Implement error component for event not found (404)
- Add placeholder for future event photos feature
- Format dates using Indonesian locale (id-ID)

rasfaxo committed:
feat(admin): create event creation form at /admin/events/new

- Build admin-only form with beforeLoad authorization check
- Add input fields: name, description, location, banner URL, date/time, visibility
- Implement real-time image preview
- Add character counters for name (max 160) and description (max 2000)
- Validate all inputs using event contracts
- Redirect to event detail page on successful creation
- Show error messages on form submission failures
- Implement UnauthorizedPage error component

rasfaxo committed:
feat(navbar): add events navigation and admin create event link

- Add /events link with Calendar icon to main navigation
- Add conditional /admin/events/new link with PlusCircle icon (admin-only)
- Maintain consistent styling with existing navigation items
- Links close sidebar on mobile after click

rasfaxo committed:
feat(events): implement server functions with validation and admin middleware

- Add getEvents and getEventById using .inputValidator and Zod schemas
- Add createEvent, updateEvent, deleteEvent protected with requireAdmin middleware
- Keep GET for reads and POST for write mutations per TanStack Start RPC pattern

rasfaxo committed:
chore(router): regenerate routeTree to include events routes

elingp reviewed (changes requested):
Should be good after this.

One more thing, updates the `seed.ts`. Add the placeholder unsplash images there. Also add another variant to the seed: an unlisted event and an event without image.

elingp commented:
I finally added the app's UI on #32. Don't forget to `rebase` and use the new UI for the Events pages.

rasfaxo committed:
fix: address PR #30 review comments

- Add conditional 'Create Event' link for admin role in DashboardLayout
- Replace Error with redirect() for unauthorized access in admin routes
- Use EventRecord type from contracts (DRY)
- Allow admin to view unlisted events with visibility badge
- Update events page to use DashboardLayout with authentication

rasfaxo committed:
feat(events): adopt new UI, show unlisted events to admin

Replace events page with latest UI (from commit 97780a9) and use real DB data
Loader: return all events for admin, keep visibility: "public" for others
Add "Unlisted" badge (EyeOff) on event cards for admin visibility
Use EventRecord from src/contracts/events.contract for card types
Move "Create event" button to /events page (admin-only)
Update event detail and create-event form to use shared UI components/styles
Seed script: add unlisted event and placeholder images (seed.ts)

rasfaxo committed:
feat(events): improve event query logic and contract types

- Update getEvents to return all events for admin (no visibility filter)
- Ensure EventRecord and related schemas/types are used consistently in contract

rasfaxo committed:
feat(event): update create event form with new UI components and validation

- Use DashboardLayout for consistent admin page layout
- Add form fields with icon labels, validation, and live character count
- Show banner image preview if URL is provided
- Add visibility select (public/unlisted)
- Use Button, Input, Label, Textarea from shared UI library
- Add redirect guard for non-admin users
- Improve error handling and UX for form submission

rasfaxo committed:
feat(events): add images and new events to seed data

rasfaxo committed:
feat(router): regenerate routeTree to include events routes

rasfaxo committed:
feat(navbar): refactor admin event link and clean up unused imports
